android/capacitor/src/main/java/com/getcapacitor/Plugin.java

```diff
@@ -283,47 +375,116 @@ public boolean hasRequiredPermissions() {
     }
 
     /**
-     * Helper for requesting a specific permission
+     * Request all of the specified permissions in the CapacitorPlugin annotation (if any)
      *
      * @since 3.0.0
      * @param call the plugin call
-     * @param permission the permission to request
-     * @param requestCode the requestCode to use to associate the result with the plugin
      */
-    public void requestPermission(PluginCall call, String permission, int requestCode) {
-        requestPermissions(call, new String[] { permission }, requestCode);
+    protected void requestAllPermissions(@NonNull PluginCall call) {
+        String callingMethodName = call.getMethodName();
+        ActivityResultLauncher<String[]> activityResultLauncher = permissionLaunchers.get(callingMethodName);
+        if (activityResultLauncher == null && !callingMethodName.equals("requestPermissions")) {
+            Logger.error(
```

Comment:
Have you thought about rejecting the call here instead?

Reply:
We could. Would it be more desirable to have a hard fail in this instance instead of falling back to a functional permission request and result (the basePermissionLauncher)? I see benefits to the fallback, but this proposal would be a more explicit indicator to the developer that something was incorrectly implemented.

Reply:
Yeah I think it might be better to fail the request so the plugin developer knows something was incorrectly implemented. (Assuming I'm understanding all the pieces at play here)